components/EmotionList.js

```diff
@@ -69,7 +69,7 @@ export default function EmotionList({
       <EmotionSection $form={form}>
         {emotionData.map(({ slug, name }) => (
           <EmotionLink
-            onClick={() => handleAddDetails(name, id)}
+            onClick={createMode && (() => handleAddDetails(name, id))}
```

Comment:
Just a short notice: For collecting data you should use a form, not a link that has an onClick. Mixing links with buttons is always bad practice. Keep this in mind at least for the future. 
Since this is a list of links you should render a `<ul>` element with `<li>`s inside.